llama-index-core/llama_index/core/llms/structured_llm.py

```diff
@@ -34,6 +34,40 @@
     OutputKeys,
     QueryComponent,
 )
+import re
+
+
+def _escape_braces(text: str) -> str:
+    """
+    Escape braces in text.
+    Only captures template variables, skips already escaped braces.
+    """
+
```

Comment:
@logan-markewich  @erik-megarad take a look? full disclosure i used claude

Reply:
testing in regex101.com, it seems like this will still escape normal template variables?

I would use the regex Eric shared earlier with this code

Reply:
Not a huge deal but without the lookahead/behind you will re-escape things that have three or more braces.

```suggestion
pattern = r"(?<!\{)\{([^{}]+?)\}(?!\})"
```

Reply:
sorry this went on longer than i expected so just summarizing the high-level objectives here to make sure we come to a quick resolution. happy to chat offline! 

- high-level goal: make `chat` work in `StructuredLLM`. `chat` expects an already-formatted list of messages
- problem: the already-formatted list of messages might contain curly brackets like {, }. due to the current implementation of `chat`, we convert the list of already-formatted messages into a `ChatPromptTemplate` in order to call `structured_predict` and output a structured output. ChatPromptTemplate treats stuff between {, } as template variables, so we need to escape any {, } to make sure that ChatPromptTemplate doesn't treat it as a template variable.

- given this, i added another test case (with 4 brackets) and added @erik-megarad 's regex

any concerns/thoughts? lmk

Reply:
Lgtm!